datafusion/core/src/datasource/physical_plan/mod.rs

```diff
@@ -482,6 +508,171 @@ fn get_projected_output_ordering(
     all_orderings
 }
 
+// A normalized representation of file min/max statistics that allows for efficient sorting & comparison.
+pub(crate) struct MinMaxStatistics {
+    min: arrow::row::Rows,
+    max: arrow::row::Rows,
+}
+
+impl MinMaxStatistics {
+    fn new_from_files<'a>(
+        sort_order: &[PhysicalSortExpr],
+        table_schema: &SchemaRef,
+        projected_schema: &SchemaRef,
+        files: impl IntoIterator<Item = &'a PartitionedFile>,
+    ) -> Result<Self> {
+        use datafusion_common::ScalarValue;
+
+        let statistics_and_partition_values = files
+            .into_iter()
+            .map(|file| {
+                file.statistics
+                    .as_ref()
+                    .zip(Some(file.partition_values.as_slice()))
+            })
+            .collect::<Option<Vec<_>>>()
+            .ok_or_else(|| {
+                DataFusionError::Plan("Parquet file missing statistics".to_string())
+            })?;
+
+        let get_min_max = |i: usize| -> (Vec<ScalarValue>, Vec<ScalarValue>) {
+            statistics_and_partition_values
+                .iter()
+                .map(|(s, pv)| {
+                    if i < s.column_statistics.len() {
+                        (
+                            s.column_statistics[i]
+                                .min_value
+                                .get_value()
+                                .cloned()
+                                .unwrap_or(ScalarValue::Null),
```

Comment:
I don't think we can just use `NULL` when the statistics are not present (as NULL will sort somewhere in the sort order, but in reality the data may be anywhere / possibly overlap) -- Thus I think this code needs to return an error in this case